integration/apps/arithmetic_intensity/build.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+#  Copyright (c) 2015 - 2021, Intel Corporation
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions
+#  are met:
+#
+#      * Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in
+#        the documentation and/or other materials provided with the
+#        distribution.
+#
+#      * Neither the name of Intel Corporation nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+set -x
+set -e
+
+# Get helper functions
+source ../build_func.sh
+
+if [ ! -x "$(command -v nasm)" ]; then
+    DIRNAME=nasm
+    GITREPO=https://github.com/netwide-assembler/nasm.git
+    TOPHASH=e2ed7b7e
+    ARCHIVE=${DIRNAME}_${TOPHASH}.tgz
+
+    get_archive ${ARCHIVE}
+    if [ -f ${ARCHIVE} ]; then
+        unpack_archive ${ARCHIVE}
+    else
+        clean_source ${DIRNAME}
+        clone_repo_git ${GITREPO} ${DIRNAME} ${TOPHASH}
+    fi
+
+    if ! ls nasm/nasm 2> /dev/null; then
+        cd nasm
+        sh autogen.sh
+        sh configure
+        make
+        cd -
+    fi
+    export PATH=$PWD/nasm:$PATH
+fi
+
+DIRNAME=ARITHMETIC_INTENSITY
+GITREPO=https://github.com/dannosliwcd/arithmetic-intensity
+TOPHASH=1d54601
+ARCHIVE=${DIRNAME}_${TOPHASH}.tgz
+
+clean_source ${DIRNAME}
+get_archive ${ARCHIVE}
+if [ -f ${ARCHIVE} ]; then
+    unpack_archive ${ARCHIVE}
+else
+    clone_repo_git ${GITREPO} ${DIRNAME} ${TOPHASH}
+fi
+setup_source_git ${DIRNAME}
+
+# build
+echo $PATH
+cd ${DIRNAME}
+make
```

Comment:
Can `make CXX=$MPICXX`, then no patch is needed.

Reply:
Done.